File: gherkin/dotnet/Gherkin/Ast/Step.cs
namespace Gherkin.Ast
{
    public class Step : IHasLocation
    {
        public Location Location { get; private set; }
        public string Keyword { get; private set; }
        public StepKeywordType KeywordType { get; }

        public string Text { get; private set; }
        public StepArgument Argument { get; private set; }

        public Step(Location location, string keyword, StepKeywordType keywordType, string text, StepArgument argument)
        {
            Location = location;
            Keyword = keyword;
            KeywordType = keywordType;
            Text = text;
            Argument = argument;
        }
    }
}

File: gherkin/java/src/main/java/gherkin/ast/Step.java
package gherkin.ast;

public class Step extends Node {
    private final String keyword;
    private final String text;
    private final Node argument;

    public Step(Location location, String keyword, String text, Node argument) {
        super(location);
        this.keyword = keyword;
        this.text = text;
        this.argument = argument;
    }

    public String getText() {
        return text;
    }

    public String getKeyword() {
        return keyword;
    }

    public Node getArgument() {
        return argument;
    }

}
